app/feats/direct_entitlement_grant_feat.py

```
@@ -0,0 +1,280 @@
+    return _execute_direct_grant_impl(
+        canonical_context=canonical_context,
+        target_seat_id=target_seat_id,
+        policy_uuid=policy_uuid,
+        quantity=quantity,
+        correlation_id=correlation_id,
+        idempotency_key=idempotency_key,
+    )
+
+
+@feat_shell("FEAT-STOR-004")
+def _execute_direct_grant_impl(
```

Ruff (0.16.0)

[warning] Too many return statements (11 > 6)

(PLR0911)